### ISSUE TITLE
spring boot delay task execution/instantiation

### ISSUE THREAD

**Issue by fuesec** [kagkarlsson/db-scheduler#208]
I have task in spring boot which gets instantiated and started like this:
```
@Configuration
public class ScheduledTasksConfiguration {

    @Bean
    public Task<Instant> myFirstTask(
            SomeService someService) {
        return recurring(
                "my-first-task",
                fixedDelay(ofMinutes(2)),
                Instant.class)
                .initialData(MIN)
                .executeStateful((taskInstance, executionContext) -> {
                    return someService.doWork(taskInstance.getData());
                });
    }
}
```

My problem is the following. Upon spring boot application startup this recurring task gets picked up and started before the Tomcat Server is fully up and running and SomeService uses a rest endpoint on its own instance (localhost:8080) for testing purposes which does not work in that case.
How can I delay the start of this task until tomcat server is running or how can I delay the task instantiation?

**Comment by kagkarlsson** [comment]
There is a property for this problem I think (see https://github.com/kagkarlsson/db-scheduler#configuration-options)

Try setting
```
db-scheduler.delay-startup-until-context-ready=true
```

**Comment by fuesec** [comment]
that seemed to do the trick. thanks a lot @kagkarlsson .

**Comment by kagkarlsson** [comment]
Perfect! We might need to consider making that the default 🤔

**Comment by kagkarlsson** [comment]
🎉 This issue has been resolved in `v16.9.0` ([Release Notes](https://github.com/kagkarlsson/db-scheduler/releases/tag/v16.9.0))
